src/main/java/com/google/devtools/build/lib/bazel/repository/starlark/StarlarkRepositoryFunction.java

```diff
@@ -240,6 +241,9 @@ private RepositoryDirectoryValue.Builder fetchInternal(
     try (Mutability mu = Mutability.create("Starlark repository")) {
       StarlarkThread thread = new StarlarkThread(mu, starlarkSemantics);
       thread.setPrintHandler(Event.makeDebugPrintHandler(env.getListener()));
+      var repoMappingRecorder = new Label.RepoMappingRecorder();
+      repoMappingRecorder.mergeEntries(rule.getRuleClassObject().getRuleDefinitionEnvironmentRepoMappingEntries());
+      thread.setThreadLocal(Label.RepoMappingRecorder.class, repoMappingRecorder);
```

Comment:
Can this be part of StarlarkThread instead? Doing thread locals this way is "action at a distance" and makes it impossible to track down what other code this affects.

Reply:
as in, you want the repo mapping recorder itself to be built into StarlarkThread? But StarlarkThread is part of the interpreter and has no concept of repo mappings or labels.

Also, I wouldn't say it's 'impossible' to track down -- it's as easy as finding all usages of `Label.RepoMappingRecorder.class`.

Reply:
I was about to say "but at least use `StarlarkThread.setThreadLocal()`,but that's not conceptually much different than just a regular thread local, so meh.

Reply:
not too sure what you mean, but I _am_ using `StarlarkThread#setThreadLocal()` :)

Reply:
Not my brightest moment, this.